tools/integration_tests/improved_run_e2e_tests.sh

```diff
@@ -75,21 +81,101 @@ PACKAGE_RUNTIME_STATS=$(mktemp "/tmp/${TMP_PREFIX}_package_stats_runtime.XXXXXX"
 RESOURCE_USAGE_FILE=$(mktemp "/tmp/${TMP_PREFIX}_system_resource_usage.XXXXXX") || { log_error "Unable to create system resource usage file"; exit 1; }
 
 # Argument Parsing and Assignments
-if [ "$#" -lt 3 ]; then
-  log_error "Missing required arguments."
-  usage
+# Set default values for optional arguments
+SKIP_NON_ESSENTIAL_TESTS_ON_PACKAGE="false"
```

Comment:
You can set and use them as booleans directly . This is how it originally was too.

```suggestion
SKIP_NON_ESSENTIAL_TESTS_ON_PACKAGE=false
```

For using,

```sh
if ${SKIP_NON_ESSENTIAL_TESTS_ON_PACKAGE}; then
  # do something
fi
```

Reply:
Same goes for other variables.